starter-repos/shared-ci-cd/appctl/prepare.yaml

```diff
@@ -17,7 +17,12 @@ prepare-manifests:
     refs:
     - master
   stage: prepare-manifests
-  image: gcr.io/cloud-solutions-images/appctl
+  image: gcr.io/cloud-solutions-images/appctl:app-delivery-demo
+  services:
+    - docker:dind
+  variables:
+    DOCKER_HOST: tcp://localhost:2375
+    DOCKER_TLS_CERTDIR: ""
```

Comment:
This requires the gitlab-runner (kubernetes executor) to run in `privileged` mode: https://docs.gitlab.com/runner/install/kubernetes.html#running-docker-in-docker-containers-with-gitlab-runners.

We need to set `runners.privileged` to true in `values.yaml` in the `gke-gitlab` module. The module loads the file from [values.yaml.tpl](https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/master/main.tf#L353). However, the template doesn't have an interpolation for the `privileged` flag: https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/master/values.yaml.tpl#L104.

I can think of two ways to enable the privileged mode:
1. update the `gke-gitlab` module with a privileged field in [values.yaml.tpl](https://github.com/terraform-google-modules/terraform-google-gke-gitlab/blob/master/values.yaml.tpl#L104) and have a local copy of the updated module in [2_gitlab](https://github.com/GoogleCloudPlatform/solutions-modern-cicd-anthos/tree/master/2_gitlab).
2. keep using the original module from the terraform registry, but add a step in cloudbuild.yaml to update the deployment by setting the `env`: 
```
kubectl set env deployments/gitlab-gitlab-runner KUBERNETES_PRIVILEGED=true
```

I'm not familiar with Terraform, but option-1 seems to add a dependency on a specific version of `gke-gitlab` module. Option-2 will cause a pod restart in the setup process.

@morgante @viglesiasce Which way would you prefer?

FYI @janetkuo

Reply:
We decided to go with adding a step in cloudbuild.yaml to avoid keeping a local copy of the terraform module.